From 764983dcdc73cb51d9cccb194f672d501e5ec9e9 Mon Sep 17 00:00:00 2001
From: Cyril Arnould <cyril.arnould@outlook.com>
Date: Tue, 13 Jun 2023 23:49:49 +0200
Subject: [PATCH 09/17] Fix infinite loops in do_time_format

In previous versions, strftime would be executed once before falling back to human_readable. The new implementation can get caught in endless loops under mingw.
---
 find/print.c | 7 +++++++
 1 file changed, 7 insertions(+)

diff --git a/find/print.c b/find/print.c
index 316a6e04..8edee57d 100644
--- a/find/print.c
+++ b/find/print.c
@@ -511,8 +511,10 @@ do_time_format (const char *fmt, const struct tm *p, const char *ns, size_t ns_s
       buf_size = 1u;
       buf = xmalloc (buf_size);
     }
+#if !defined(__MINGW32__) || defined(_UCRT)
   while (true)
     {
+#endif      
       /* I'm not sure that Solaris will return 0 when the buffer is too small.
        * Therefore we do not check for (buf_used != 0) as the termination
        * condition.
@@ -567,9 +569,14 @@ do_time_format (const char *fmt, const struct tm *p, const char *ns, size_t ns_s
         }
       else
         {
+#if defined(__MINGW32__) && !defined(_UCRT)
+          return 0;
+        }
+#else
           buf = x2nrealloc (buf, &buf_size, sizeof *buf);
         }
     }
+#endif      
 }
 
 /* Return a static string formatting the time WHEN according to the
-- 
2.41.0

